lib/binary_search_tree.rb

```diff
@@ -8,28 +8,47 @@ def insert(score, title)
 
     return set_root(new_node) if root.nil?
 
-    current_node = root
-    current_depth = 0
-
-    until current_node.movie_score == score
-      if score < current_node.movie_score
-        current_node.left = new_node if current_node.left.nil?
-        current_node = current_node.left
-      elsif score > current_node.movie_score
-        current_node.right = new_node if current_node.right.nil?
-        current_node = current_node.right
-      end
+    parent = find_node_or_parent_for(score)
+    return parent.depth if score == parent.movie_score
 
-      current_depth += 1
-    end
-    
-    current_depth
+    new_node.depth = parent.depth + 1
+    score < parent.movie_score ?
+      parent.left = new_node :
+      parent.right = new_node
+
+    new_node.depth
+  end
+
+  def include?(score)
+    node = find_node_or_parent_for(score)
+    score == node.movie_score
+  end
+
+  def depth_of(score)
+    node = find_node_or_parent_for(score)
+    return node.depth if node.movie_score == score
   end
```

Comment:
These look very similar... Though they are very small, i don't think it is necessary to pull out the functionality.